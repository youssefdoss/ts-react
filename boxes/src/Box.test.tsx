import { render } from '@testing-library/react';
import Box from './Box';

describe('Box component', function (): void {
  test('renders without crashing', function (): void {
    render(
      <Box
        id='test'
        width='20'
        height='20'
        backgroundColor='greenyellow'
        remove={jest.fn()}
      />
    );
  });

  test('matches snapshot', function (): void {
    const { container } = render(
      <Box
        id='test'
        width='20'
        height='20'
        backgroundColor='greenyellow'
        remove={jest.fn()}
      />
    );

    expect(container).toMatchSnapshot();
  });
});