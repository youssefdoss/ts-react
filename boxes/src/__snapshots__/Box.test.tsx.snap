// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component matches snapshot 1`] = `
<div>
  <div
    class="Box"
  >
    <div
      class="Box-box"
      style="height: 20em; width: 20em; background-color: greenyellow;"
    />
    <button
      class="Box-removeBtn"
    >
      Remove The Box!
    </button>
  </div>
</div>
`;
